src/core/execute-serialize.c

```diff
@@ -1606,17 +1708,16 @@ static int exec_parameters_deserialize(ExecParameters *p, FILE *f, FDSet *fds) {
                                 if (r == 0)
                                         break;
 
-                                if ((fd = parse_fd(w)) < 0 || !fdset_contains(fds, fd))
-                                        log_debug("Failed to parse %s value: %s, ignoring.", l, w);
-                                else {
-                                        r = fdset_remove(fds, fd);
-                                        if (r < 0) {
-                                                log_debug_errno(r, "Failed to remove %s value=%d from fdset, ignoring: %m", l, fd);
-                                                continue;
-                                        }
-
-                                        p->fds[i] = fd;
+                                if (startswith(l, "exec-parameters-fds-by-fd-index="))
+                                        fd = deserialize_fd_from_array(w, fds_array, n_fds_array);
+                                else
+                                        fd = deserialize_fd_from_set(w, fds);
+                                if (fd < 0) {
+                                        log_debug_errno(fd, "Failed to deserialize %s value: %s, ignoring.", l, w);
```

Comment:
sorry, but this is a mess, please don#t copy this all over the place. 

seriously distinguish the two cases in the field value, so that you don#t have to duplicate *every* *single* fd deserialization in two branches.

Reply:
commit dropped